helm-chart-validation/action.yml

```diff
@@ -34,9 +37,10 @@ runs:
       uses: greenbone/actions/set-github-user@v3
 
     - name: Clone Repository
+      if: inputs.repository-url && inputs.repository-branch
       shell: bash
       run: |
-        git clone ${{ inputs.repository-url }}
+        git clone --branch ${{ inputs.repository-branch }} ${{ inputs.repository-url }}
```

Comment:
Use the [`actions/checkout` action](https://github.com/actions/checkout) for this. ;)